**Print Helloworld**

In [3]:
print("Hello")
print("Hello")
10

Hello
Hello


10

**Variables**

In [5]:
a = 10
b = 20

a + b
sum = a + b
abc=10
print(sum)
abc

30


10

- without giving print only we got output 30.
- This is output of 2nd cell .
- There are 2 ways by which we print
- 1)Is we use print command
- 2)whatever is lastline the output or result of that.
- print command if you want to print a string like text command you need - - to put in single or double quote.-->This prints sum.
- But If you want to print storage we can simply say sum. -->this gives - - value inside storage sum.

- **How to name variables**
- Variables can have any character either lower case or upper case or - Digit or underscore.
- **The only restriction is you cannot start a variable name with digit.**
- The variable can start with upper or lower case or underscore.


- we know how to create variables access them later and we know how we name them as well.
- Next we will look how exactly we are storing the data.

- Now lets continue our discussion on variables.
- If we give # infront of code it gets commented.
- Its just like if we want to add some notes in front my code.





**Assigning different type of data to a variable**

- In java we need to give upfront what kind of data you want to store in a.
If you want to store integer you need to give int a
That basically means we are telling java that i need a storage uprfont where i need to store integer.
That not case with python.
- Python allows it but java doesnt what is the difference

- Thats where we need to know how things are stored in python and it is different from java.
- In java when we say int a =10 we give fixed amount of space for integer
you can put what ever value you want to put within 4 bytes.
we give int upfront in java because java is deciding how much space is required upfront.It is giving space ased on what kind of data you want to store.
- In case of python things are different when we say a=10 it doesnt mean a is fixed storage where 10 is going to be stored.
what happens is it creates some storage which is 10 and then a just stores address of this storage.

- Both are diiferent because a=20 in next line if we give python will create a different storage which has 20 in it.
Instead of a address it will start storing 20 address.

- The difference is in java one storage is dedicated to a and any storage you put inside a comes here.

- In python a is storing in address of some storage it can be storage contating 10 or 20.
- Because you are doing this even if you give a="asf" pyhton will create new storage with all this written in it.
- And a will now contain this new storage instead of previous one.



- Lets see again In python 
when we give a=10 it contains adress of the storage
when we give a=20 it contains new storage and a will start storing new address instead of previous one.
a is actually address of some storage
Next time a="abc" a will store this address instead of previous one.


- where as when we say anything a contains is going to contain within 4 bytes we are restricting ourself to what all we can store in a.
In python we can store integer as well strings.

- There is intresting command or function called type.
Using which you can figure out what kind of data is currently stored in it.

In [7]:
type(a)

int

In [8]:
a = "abc"
print(type(a))
a = 10
print(type(type(a)))

<class 'str'>
<class 'type'>


## Python Numbers

we can have 3 types as below integers,floating point numbers and complex numbers.

In [10]:
a1 = 23
a2 = 3.4
a3 = 4 + 5j
print(type(a1))
print(type(a2))
print(type(a3))

<class 'int'>
<class 'float'>
<class 'complex'>


In [12]:
a = 10
print(id(a))
a = a + 1
print(id(a))
a

94103465036576
94103465036608


11

- what exactly is happening? a =10 python created some storage at 10 and also has some metadata some additional information.
And contains address of this place.
- Next line we side a = a + 1 so firstly pyhton evaluates a + 1 which is 11.So it has storage which contains 11 some metadata here as well.
And now a instead of having previous address will have address of storage 11.
So address itself has changed.
So when we do this stogare is changed.
we have another instreting command called id command.
- If we give Id of some variable it will give unique number associated with storage that this number is pointing to.
- If we print id(a) we see 2 different numbers.
- This basically means a is ponting to some storage but after we did a = a + 1 a is pointing to some other storage.
- Id method basically gives you unique number associated with the storage that a variable is storing the address off.




In [13]:
a=10
b=10
print(id(a))
print(id(b))

94103465036576
94103465036576


- Both a and b have same id.Meaning both a and b are storing the address of same storage.
- There is one storage 10 in it and both a and b are storing the addresses.

In [14]:
a=1000
b=1000
print(id(a))
print(id(b))


140471801144880
140471801146160


Now if we see for 1000 the ids are different.so python is doing optimization but does only for numbers between -5 to 256.
- For all numbers within this range if 2 variables are storing the same numbers they will be pointing or address of same storage.
- But numbers outside this the optimization is not done.

In [16]:
a =10
id1=id(a)
b=a+2-2
id2=id(b)
print(id(a))
print(id(b))

94103465036576
94103465036576


# Arithmatic Operators

Arithmatic Operatiors + - = / // ** % 
Lets talk about these 7 operators


In [20]:
a = 10
b = 4
print(a-b)
print(a+b)

6
14


In [21]:
a * b 

40

In [22]:
a/b # Floating point division

2.5

In [23]:
a//b # Integer division

2

In [26]:
a % b # Remainder

2

In [25]:
a**b # Exponentiation

10000

- a/b gives decimal 
- a//b gives only integer part


we need to take care what happens 1st and what happens later.


In [27]:
2 + 3 * 4 # 14 or 20

14

In [29]:
2 * 3 / 4 # Doesnt matter what operation is done first

1.5

Put brackets if you want to do that operation 1st
For SI also give brackets.

In [30]:
2 * 3 // 4 # Multiplication 1st and then division

1

In [34]:
# Simple Interest to lower integer
p = 100
r = 12
t = 2
SI = p*t*r//100
SI

24

In [35]:
# Simple Interest
p = 100
r = 12
t = 2
SI =t//100*r*p
SI

0

In [41]:
# fah to celcius conversion
f = 100
c = (f-32) * 5/9
print(c)
ci =(f-32) * 5//9  # Integer celsius value
print(ci)

37.77777777777778
37


# Taking Inputs
- We want our code to ask user the 3 numbers for SI when we give 3 numbers then we would want to do calculation and figure out the answer.
- Lets write a simple code to add 2 numbers


## input

- If we write input the code is running and waiting for the user.
- And if we give 23 it gives us 23 as output.
- So we want to store this number thats what we want to do right.


In [43]:
input()

23


'23'

In [44]:
a = input()
a

23


'23'

In [47]:
a = input()
b = input()
print(type(a))
s = a + b
print(s)

23
34
<class 'str'>
2334


- we are hoping for 57 answer but we got 2334 as answer.
what this means is there is somthing wrong in the way it has treated a and b.
- a is string 23 and b is string 34 when it adds 2 strings we get 2334.
- It is adding 2 strings it is taking input as integer.
- The input command has given a string irrespective of user has given.
- Lets try to convert string 23 into number 23.
- To convert string to integer we use basically command int('34') it will convert into integer.
- we can use int to convert string or floating numbers to integers.
- If it is not a valid integer you will get a error.

In [50]:
#a = input()
#b = input()
b = '34'
c = int(b)
print(type(c))
print(c)

<class 'int'>
34


In [51]:
a = int(input())
print(type(a))
b = int(input(a))
print(type(b))
s = a + b
print(s)

23
<class 'int'>
2334
<class 'int'>
57


- It will only for integers in case you want to take floating point as input you can use float instead of int.


In [52]:
a = float(input())
print(type(a))
b = float(input(a))
print(type(b))
s = a + b
print(s)

3.5
<class 'float'>
3.53.6
<class 'float'>
7.1


- To take input we just need to use input command the input command will result in string to convert to integer we need to use int command  and to convert into float we need to use the float command.

In [53]:
a = int(input())
b = int(input())
c = a + b
print(c)

abc


ValueError: ignored